test/contracts/regression/AdminUpgradeabilityProxy.test.js

```diff
@@ -0,0 +1,122 @@
+    await proxyInstance1.setValue(42);
+
+    const instance2 = await Implementation2.new();
+    await proxy.upgradeTo(instance2.address, {
+      from: admin
+    });
+
+    const proxyInstance2 = Implementation2.at(proxy.address);
+    const res = await proxyInstance2.getValue.call();
+    assert.equal(res.toString(), "42");
+  });
+
+  it('should remove function', async () => {
+    const instance2 = await Implementation2.new();
+
+    const proxy = await AdminUpgradeabilityProxy.new(instance2.address, {
+      from: admin
+    });
+
+    const proxyInstance2 = await Implementation2.at(proxy.address);
+    await proxyInstance2.setValue(42);
+    const res = await proxyInstance2.getValue.call();
+    assert.equal(res.toString(), "42");
+
+    const instance1 = await Implementation1.new();
+
+    await proxy.upgradeTo(instance1.address, {
+      from: admin
+    });
+
+    const proxyInstance1 = await Implementation2.at(proxy.address);
+    assertRevert(proxyInstance1.getValue.call());
+  });
+
+  it('should change function signature', async () => {
+    const instance1 = await Implementation1.new();
+
+    const proxy = await AdminUpgradeabilityProxy.new(instance1.address, {
+      from: admin
+    });
+
+    const proxyInstance1 = await Implementation1.at(proxy.address);
+    await proxyInstance1.setValue(42);
+
+    const instance3 = await Implementation3.new();
+    await proxy.upgradeTo(instance3.address, {
+      from: admin
+    });
+
+    const proxyInstance3 = Implementation3.at(proxy.address);
+    const res = await proxyInstance3.getValue.call(8);
+    assert.equal(res.toString(), "50");
+  });
+
+  it('should add fallback function', async () => {
+    const instance1 = await Implementation1.new();
+
+    const proxy = await AdminUpgradeabilityProxy.new(instance1.address, {
+      from: admin
+    });
+    const proxyInstance1 = await Implementation1.at(proxy.address);
+
+    const instance4 = await Implementation4.new();
+    await proxy.upgradeTo(instance4.address, {
+      from: admin
+    });
+
+    const proxyInstance4 = await Implementation4.at(proxy.address);
+
+    await sendTransaction(proxy, '', [], [], {
+      from: anotherAddress
+    });
+
+    const res = await proxyInstance4.getValue.call();
+    assert.equal(res.toString(), "1");
+  });
+
+  it('should remove fallback function', async () => {
+    const instance4 = await Implementation4.new();
+    const proxy = await AdminUpgradeabilityProxy.new(instance4.address, {
+      from: admin
+    });
+    const proxyInstance4 = await Implementation4.at(proxy.address);
+
+    const instance2 = await Implementation2.new();
+    await proxy.upgradeTo(instance2.address, {
+      from: admin
+    });
+
+    await assertRevert(sendTransaction(proxy, '', [], [], {
+      from: anotherAddress
+    }));
+
+    const proxyInstance2 = Implementation2.at(proxy.address);
+
+    const res = await proxyInstance2.getValue.call();
+    assert.equal(res.toString(), "0");
+  });
+});
```

Comment:
Nit, but I would move this tests into a `'regression'` context within the `contracts/upgradeability/AdminUpgradeabilityProxy.test.js` file